base/initdefs.jl

```diff
@@ -214,6 +214,10 @@ function init_load_path()
     project = (JLOptions().project != C_NULL ?
         unsafe_string(Base.JLOptions().project) :
         get(ENV, "JULIA_PROJECT", nothing))
+    if project !=nothing && occursin('~', project)
+        project = project[findlast(isequal('~'), project):end]
```

Comment:
What's the purpose of this? Why not just call `expanduser` which returns its argument as is if it doesn't start with a `~`?

Reply:
Ok, I think I see what's going on here. Let me know if I'm correct. Earlier a relative path is turned absolute before user expansion, which leaves a tilde in the middle of the path and this applies the user expansion after a tilde occurs.

So this is not the right way to deal with that because if the user actually did `--project=/tmp/~/Documents` then we should not user expand that. So there's a few ways to go about this. You could make paths absolute later on, after doing user expansion. Or you could do user expansion earlier, before making paths absolute. I can't recall if there was a reason that making paths absolute had to happen so early in the boot process, but assuming not, then I would favor doing the absolute path conversion later on at this point in the boot process.

Reply:
right, as I mentioned in the issue, `Base.JLOptions().project` has expanded the current path so if we're in `/tmp` then the `--project` will be appended resulting `/tmp/~/Documents` and such.